src/helperFunctions/program_setup.py

```diff
@@ -64,40 +39,59 @@ def _setup_argparser(name, description, command_line_options, version=__VERSION_
     parser.add_argument('-V', '--version', action='version', version=f'{name} {version}')
     parser.add_argument('-l', '--log_file', help='path to log file', default=None)
     parser.add_argument(
-        '-L', '--log_level', help='define the log level', choices=['DEBUG', 'INFO', 'WARNING', 'ERROR'], default=None
+        '-L',
+        '--log_level',
+        help='define the log level for the console',
+        choices=['DEBUG', 'INFO', 'WARNING', 'ERROR'],
+        default='INFO',
     )
-    parser.add_argument('-d', '--debug', action='store_true', default=False, help='print debug messages')
-    parser.add_argument('-s', '--silent', action='store_true', default=False, help='don\'t log to command line')
-    parser.add_argument('-C', '--config_file', help='set path to config File', default=f'{get_config_dir()}/main.cfg')
+    parser.add_argument('-C', '--config_file', help='set path to config File', default=None)
     parser.add_argument(
         '-t', '--testing', default=False, action='store_true', help='shutdown system after one iteration'
     )
     parser.add_argument('--no-radare', default=False, action='store_true', help='don\'t start radare server')
     return parser.parse_args(command_line_options[1:])
 
 
-def setup_logging(args, component=None):
-    log_level = getattr(logging, cfg.logging.loglevel, None)
+def get_logging_config(args, component):
+    """
+    Returns a tuple of (logfile, loglevel) read from args and the config file.
+    The loglevel is returned as an integer.
+    Assumes that `config.load` was called beforehand.
+    """
+    console_loglevel = logging.getLevelName(args.log_level)
+    if isinstance(console_loglevel, str):
+        raise Exception(f'Invalid log level: {console_loglevel}')
```

Comment:
I don't think this is even necessary as the possible values for this are limited by the "choices" of the argument in argparse (meaning this should always result in a valid log level).

Reply:
Good catch.